src/pages/studyView/StudyViewPage.tsx

```diff
@@ -5,70 +5,135 @@ import $ from 'jquery';
 import {observer} from "mobx-react";
 import {remoteData} from "../../shared/api/remoteData";
 import {action, computed, observable, reaction} from "mobx";
-import client from "shared/api/cbioportalInternalClientInstance";
+import internalClient from "shared/api/cbioportalInternalClientInstance";
+import defaultClient from "shared/api/cbioportalClientInstance";
+import { ClinicalDataCountFilter, StudyViewFilter, ClinicalDataCount, ClinicalDataEqualityFilter } from 'shared/api/generated/CBioPortalAPIInternal';
+import { PieChart } from './charts/pieChart/PieChart'
+import { ClinicalAttribute } from 'shared/api/generated/CBioPortalAPI';
+import styles from "./styles.module.scss";
 
 
-function getMutatedGenes(query:string){
-    console.log("fetching genes");
-    return new Promise((resolve,reject)=>{
-        setTimeout(function(){
-            resolve(_.filter(["aaron","nora","natasha", "john"],(name:string)=>name.includes(query)))
-        },1000);
-    });
-
-}
-
-
-// this is an example of using client.
-client.getAllGenesetsUsingGET({}).then(function(data){
-    console.log(data);
-});
-
-class StudyViewPageStore {
+export type ClinicalAttributeData = {[attrId:string]:ClinicalDataCount[]};
+export type ClinicalAttributeDataWithMeta = {attributeId:string,clinicalDataType:"SAMPLE" | "PATIENT",counts:ClinicalDataCount[]};
+export class StudyViewPageStore {
 
     constructor(){
 
     }
 
-    @observable filterTerm = "";
-
-    readonly mutatedGenes = remoteData({
-
-        await:()=>[
-            this.prelimData
-        ],
-
-        invoke:()=>{
-
-            return getMutatedGenes(this.filterTerm);
-
+    @observable studyId = "hnsc_tcga"
+
+    @observable sampleAttrIds = ["OCT_EMBEDDED"]
+
+    @observable patientAttrIds = ["AJCC_PATHOLOGIC_TUMOR_STAGE","ICD_O_3_HISTOLOGY"]
+    
+    @observable filter:StudyViewFilter;
+
+    @observable propFilters:{[id:string]:ClinicalDataEqualityFilter} = {}
```

Comment:
You probably want @observable.ref  (look it up in mobx documentation).  observable will observe deeply (into properties of filters.  unless those actually change atomically (i.e. you mutate the "leaves" of the filter but leave the filter unchanged, you don't need deep observation because filter is replaced whole every time it's updated.

Reply:
let me know if you need more explaination.  callin @observer is NOT costly.  but calling @observable IS costly so we should try to limit deep observation when we don't need i.t